modulemd/v2/modulemd-translation-entry.c

```diff
@@ -0,0 +1,623 @@
+/*
+ * This file is part of libmodulemd
+ * Copyright (C) 2018 Red Hat, Inc.
+ *
+ * Fedora-License-Identifier: MIT
+ * SPDX-2.0-License-Identifier: MIT
+ * SPDX-3.0-License-Identifier: MIT
+ *
+ * This program is free software.
+ * For more information on the license, see COPYING.
+ * For more information on free software, see <https://www.gnu.org/philosophy/free-sw.en.html>.
+ */
+
+#include <glib.h>
+#include <yaml.h>
+
+#include "modulemd-translation-entry.h"
+#include "private/glib-extensions.h"
+#include "private/modulemd-translation-entry-private.h"
+#include "private/modulemd-util.h"
+#include "private/modulemd-yaml.h"
+
+#define TE_DEFAULT_STRING "__LOCALE_UNSET__"
+
+struct _ModulemdTranslationEntry
+{
+  GObject parent_instance;
+
+  gchar *locale;
+  gchar *summary;
+  gchar *description;
+
+  GHashTable *profile_descriptions;
+};
+
+G_DEFINE_TYPE (ModulemdTranslationEntry,
+               modulemd_translation_entry,
+               G_TYPE_OBJECT)
+
+enum
+{
+  PROP_0,
+
+  PROP_LOCALE,
+  PROP_SUMMARY,
+  PROP_DESCRIPTION,
+
+  N_PROPS
+};
+
+static GParamSpec *properties[N_PROPS];
+
+
+ModulemdTranslationEntry *
+modulemd_translation_entry_new (const gchar *locale)
+{
+  // clang-format off
+  return g_object_new (MODULEMD_TYPE_TRANSLATION_ENTRY,
+                       "locale", locale,
+                       NULL);
+  // clang-format on
+}
+
+
+ModulemdTranslationEntry *
+modulemd_translation_entry_copy (ModulemdTranslationEntry *self)
+{
+  g_autoptr (ModulemdTranslationEntry) te = NULL;
+  g_return_val_if_fail (MODULEMD_IS_TRANSLATION_ENTRY (self), NULL);
+
+  te = modulemd_translation_entry_new (
+    modulemd_translation_entry_get_locale (self));
+
+  modulemd_translation_entry_set_summary (
+    te, modulemd_translation_entry_get_summary (self));
+  modulemd_translation_entry_set_description (
+    te, modulemd_translation_entry_get_description (self));
+
+  g_hash_table_unref (te->profile_descriptions);
+  te->profile_descriptions =
+    _modulemd_hash_table_deep_str_copy (self->profile_descriptions);
+
+  return g_object_ref (te);
+}
+
+
+static void
+modulemd_translation_entry_finalize (GObject *object)
+{
+  ModulemdTranslationEntry *self = (ModulemdTranslationEntry *)object;
+
+  g_clear_pointer (&self->locale, g_free);
+  g_clear_pointer (&self->summary, g_free);
+  g_clear_pointer (&self->description, g_free);
+
```

Comment:
You aren't freeing the profiles here. That's a memory leak. The valgrind test should have caught that... I wonder why it did not.